C7FIT/AbstractHealthCell.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  AbstractHealthCell.swift
+//  C7FIT
+//
+//  Created by Brandon Lee on 1/31/17.
+//  Copyright © 2017 Brandon Lee. All rights reserved.
+//
+
+import UIKit
+
+open class AbstractHealthCell: UITableViewCell {
+    
+    // MARK: - Constants
+    
+    open let picker = UIPickerView()
+    
+    // MARK: - Properties
+    
+    var dataTitle: UILabel = UILabel()
+    var dataLabel: UILabel = UILabel()
+    
+    open weak var delegate: PickerCellDelegate?
+    open weak var dataSource: PickerCellDataSource?
+
+    // MARK: - Initialization
+    
+    public override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        picker.delegate = self
+        picker.dataSource = self
+        setup()
+        setupConstraints()
+    }
+    
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - UIPickerView
+    
+    open override var canBecomeFirstResponder: Bool {
+        return true
+    }
+    
+    open override var canResignFirstResponder: Bool {
+        return true
+    }
+    
+    open override func becomeFirstResponder() -> Bool {
+        picker.dataSource = self
+        delegate?.onPickerOpen(self)
+        return super.becomeFirstResponder()
+    }
+    
+    open override func resignFirstResponder() -> Bool {
+        delegate?.onPickerClose(self)
+        return super.resignFirstResponder()
+    }
+    
+    open override var inputView: UIView? {
+        return picker
+    }
+    
+    // MARK: - Layout
+    
+    open func setup() {
+        dataTitle.backgroundColor = .red
+        addSubview(dataTitle)
+        
+        dataLabel.backgroundColor = .cyan
+        addSubview(dataLabel)
+    }
+    
+    open func setupConstraints() {
+        dataTitle.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
How does that work? I tried replacing my code my let constants and `NSLayoutConstraint.activate()` with your sample code but Xcode is complaining about unresolved identifiers?